rules/REQUEST-944-APPLICATION-ATTACK-JAVA.conf

```diff
@@ -142,6 +142,45 @@ SecRule ARGS|ARGS_NAMES|REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES
     setvar:'tx.anomaly_score_pl1=+%{tx.critical_anomaly_score}'"
 
 
+#
+# [ Java Script Uploads ]
+#
+# Block file uploads with filenames ending in Java scripts (.jsp, .jspx)
+#
+# Many application contain Unrestricted File Upload vulnerabilities.
+# https://www.owasp.org/index.php/Unrestricted_File_Upload
+#
+# Attackers may use such a vulnerability to achieve remote code execution
+# by uploading a script file. If the upload storage location is predictable
+# and not adequately protected, the attacker may then request the uploaded
+# file and have the code within it executed on the server.
+#
+# Some AJAX uploaders use the nonstandard request headers X-Filename,
+# X_Filename, or X-File-Name to transmit the file name to the server;
+# scan these request headers as well as multipart/form-data file names.
+#
+SecRule FILES|REQUEST_HEADERS:X-Filename|REQUEST_HEADERS:X_Filename|REQUEST_HEADERS:X.Filename|REQUEST_HEADERS:X-File-Name "@rx .*\.(?:jsp|jspx)\.*$" \
+    "id:944140,\
+    phase:2,\
+    block,\
+    capture,\
+    t:none,t:lowercase,\
+    msg:'Java Injection Attack: Java Script File Upload Found',\
+    logdata:'Matched Data: %{TX.0} found within %{MATCHED_VAR_NAME}: %{MATCHED_VAR}',\
+    tag:'application-multi',\
+    tag:'language-java',\
+    tag:'platform-multi',\
+    tag:'attack-injection-java',\
+    tag:'paranoia-level/1',\
+    tag:'OWASP_CRS',\
+    tag:'capec/1000/152/242',\
+    ctl:auditLogParts=+E,\
+    ver:'OWASP_CRS/3.4.0-dev',\
+    severity:'CRITICAL',\
+    setvar:'tx.rce_score=+%{tx.critical_anomaly_score}',\
```

Comment:
Is this an rce? 🤔

Reply:
Do we have a better category? If you manage to upload a .jsp into an executable dir you achieve RCE.

Reply:
No, you are right. Let's stick with rce.